upstairs/src/in_memory.rs

```diff
@@ -57,28 +57,29 @@ impl BlockIO for InMemoryBlockIO {
     async fn read(
         &self,
         offset: Block,
-        data: Buffer,
-    ) -> Result<(), CrucibleError> {
+        mut data: Buffer,
+    ) -> Result<Buffer, CrucibleError> {
+        // Read from self into Buffer
         let inner = self.inner.lock().await;
 
-        let mut data_vec = data.as_vec().await;
-        let mut owned_vec = data.owned_vec().await;
-
         let start = offset.value as usize * self.block_size as usize;
 
-        for i in 0..data_vec.len() {
-            data_vec[i] = inner.bytes[start + i];
-            owned_vec[i] = inner.owned[start + i];
-        }
+        data.write_with_ownership(
+            0,
+            &inner.bytes[start..(start + data.len())],
+            &inner.owned[start..(start + data.len())],
+        );
 
-        Ok(())
+        Ok(data)
     }
 
     async fn write(
         &self,
         offset: Block,
         data: Bytes,
     ) -> Result<(), CrucibleError> {
+        // Write from Buffer into self
```

Comment:
Want to elevate this to a docstring, e.g.
```
/// Write from `data` to `self`, setting all owned bits to `true`
```

Reply:
done in 4b89d18